backend/src/app.module.ts

```diff
@@ -10,19 +10,29 @@ import { AuthMiddleware } from './common/middleware/auth.middleware';
 import { JwtModule } from '@nestjs/jwt';
 import { ChatController } from './chat/chat.controller';
 import { ChatService } from './chat/chat.service';
+import { MongooseModule } from '@nestjs/mongoose';
+import {User, UserSchema} from './db/schemas/user.schema';
+
+// TODO: move this to a .env file 
+const MONGO_URL = 'mongodb://localhost/';
+const MONGO_DB = 'nestjs-chatapp';
 
 @Module({
+  // TODO: move JWT secret to a .env file
   imports: [
     JwtModule.register({ secret: process.env.JWT_SECRET || 'jwt_secret' }),
     ConfigModule.forRoot({
       isGlobal: true,
     }),
+    MongooseModule.forRoot(MONGO_URL+MONGO_DB),
+    // TODO: It's possible to move this line to the corresponding submodule which deals with the specific logic. Currently we have one module - App module for handling everything
+    MongooseModule.forFeature([{ name: User.name, schema: UserSchema }])
   ],
   controllers: [
     AppController,
     UserController,
     MessageController,
-    ChatController,
+    ChatController
```

Comment:
This will keep coming back whenever I commit changes, as the VSCode default NestJS linter does this. We can get a husky or better linter config to avoid this.